File: VB/WinWebSolution.Module/NestedListViewControllerBase.vb
Imports System
Imports System.Reflection
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions

Namespace WinWebSolution.Module
    Public MustInherit Class NestedListViewControllerBase
        Inherits ViewController
        Implements IMasterDetailViewInfo

        Public Sub New()
            TargetViewNesting = Nesting.Nested
            TargetViewType = ViewType.ListView
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
        End Sub
        Private masterDetailViewIdCore As String = String.Empty
        #Region "IMasterDetailViewInfo Members"
        Public ReadOnly Property MasterDetailViewId() As String Implements IMasterDetailViewInfo.MasterDetailViewId
            Get
                Return masterDetailViewIdCore
            End Get
        End Property
        Public Sub AssignMasterDetailViewId(ByVal id As String) Implements IMasterDetailViewInfo.AssignMasterDetailViewId
            masterDetailViewIdCore = id
        End Sub
        #End Region
    End Class
End Namespace

File: CS/WinWebSolution.Module/Controllers/NestedListViewControllerBase.cs
using System;
using System.Reflection;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;

namespace WinWebSolution.Module {
    public abstract class NestedListViewControllerBase : ViewController, IMasterDetailViewInfo {
        public NestedListViewControllerBase() {
            TargetViewNesting = Nesting.Nested;
            TargetViewType = ViewType.ListView;
        }
        protected override void OnActivated() {
            base.OnActivated();
        }
        private string masterDetailViewIdCore = String.Empty;
        #region IMasterDetailViewInfo Members
        public string MasterDetailViewId {
            get { return masterDetailViewIdCore; }
        }
        public void AssignMasterDetailViewId(string id) {
            masterDetailViewIdCore = id;
        }
        #endregion
    }
}
